gnomad_constraint/constraint/constraint_pipeline.py

```diff
@@ -0,0 +1,65 @@
+# noqa: D100
+
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.resources.grch37.gnomad import public_release
+from gnomad_constraint.resources.resource_utils import (
+    get_processed_ht_path,
+    get_logging_path,
+    context_ht_path,
+)
+from gnomad_constraint.utils.constraint_basics import (
+    add_vep_context_annotations,
+    prepare_ht_for_constraint_calculations,
+)
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("constraint_pipeline")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """Execute the constraint pipeline."""
+    try:
+        if args.add_vep_annotations:
+            logger.info("Adding VEP context annotations...")
+            # TODO: Need to add function that annotates methylation, coverage, and gerp in the vep context table.
+            add_vep_context_annotations(
+                public_release("genomes").ht(), context_ht_path
+            ).write(get_processed_ht_path("genomes"), overwrite=args.overwrite)
+            add_vep_context_annotations(
+                public_release("exomes").ht(), context_ht_path
+            ).write(get_processed_ht_path("exomes"), overwrite=args.overwrite)
+            logger.info("Done with preprocessing genome and exome Table.")
+
+        full_context_ht = prepare_ht_for_constraint_calculations(
```

Comment:
Since `context_ht` and `exome_ht` are reused a lot in the downstream, is it necessary to move the checkpoint for `add_vep_context_annotations ` to here or where we defined `context_ht`?